src/next/azext_next/custom.py

```diff
@@ -2,60 +2,97 @@
     if not recommends:
         send_feedback(request_type, -1, command_history, processed_exception)
         print("\nSorry, there is no recommendation in the next step.")
         return
 
     print()
-    _give_recommends(cmd, recommends)
 
-    option = get_int_option("Please select your option " + Fore.LIGHTBLACK_EX + "(if none, enter 0)" +
-                            Fore.RESET + ": ", 0, len(recommends), -1)
-    if option == 0:
-        send_feedback(request_type, 0, command_history, processed_exception, recommends)
-        print('\nThank you for your feedback. If you have more feedback, please submit it by using "az feedback" \n')
-        return
-    print()
+    # divide recommendation items into two sets
+    command_recommendations = [item for item in recommends if item['type'] != RecommendType.Scenario]
+    scenario_recommendations = [item for item in recommends if item['type'] == RecommendType.Scenario]
+    has_multi_type_recommendation = command_recommendations and scenario_recommendations
+
+    if not has_multi_type_recommendation:
+        _give_recommends(cmd, recommends)
+
+        option_msg = [(Style.PRIMARY, "Please select your option "),
+                      (Style.SECONDARY, "(if none, enter 0)"), (Style.PRIMARY, ": ")]
+        option = select_option(option_msg, min_option=0, max_option=len(recommends), default_option=-1)
+        if option == 0:
+            send_feedback(request_type, 0, command_history, processed_exception, recommends)
+            print(
+                '\nThank you for your feedback. If you have more feedback, please submit it by using "az feedback" \n')
+            return
+        print()
 
-    rec = recommends[option - 1]
-    send_feedback(request_type, option, command_history, processed_exception, recommends, rec)
+        rec = recommends[option - 1]
+        send_feedback(request_type, option, command_history, processed_exception, recommends, rec)
+    else:
+        # display scenario recommendations with prefix 'b', and command recommendations with prefix 'a'
+        print_styled_text([(Style.PRIMARY, "COMMAND")])
+        print()
+        _give_recommends(cmd, command_recommendations, prefix='a')
+        print_styled_text([(Style.PRIMARY, "SCENARIO")])
+        print()
+        _give_recommends(cmd, scenario_recommendations, prefix='b')
+        option_msg = [(Style.PRIMARY, "Please select your option "),
+                      (Style.SECONDARY, "(for example, enter \"b2\" for the second scenario. if none, enter 0)"),
+                      (Style.PRIMARY, ": ")]
+        group, option = select_combined_option(
+            option_msg,
+            {'a': OptionRange(1, len(command_recommendations)), 'b': OptionRange(1, len(scenario_recommendations))},
+            (None, -1))
+        if group == 'a':
+            rec = command_recommendations[option - 1]
+        elif group == 'b':
+            rec = scenario_recommendations[option - 1]
```

Comment:
We also need to send feedback data in these two cases

Reply:
Done